compiler/luci/service/src/Nodes/CirclePad.cpp

```diff
@@ -32,11 +32,24 @@ namespace sinf
 
 loco::TensorShape Algorithm::visit(const luci::CirclePad *node)
 {
-  // TODO support non-const case
-  auto paddings = loco::must_cast<luci::CircleConst *>(node->paddings());
+  auto paddings = dynamic_cast<luci::CircleConst *>(node->paddings());
   auto circle_input = loco::must_cast<const luci::CircleNode *>(node->input());
   auto input_shape = circle_shape(circle_input);
-  return pad_shape(input_shape, paddings);
+  if (paddings != nullptr)
+  {
+    return pad_shape(input_shape, paddings);
+  }
+  else
+  {
+    loco::TensorShape output_shape;
+    int32_t n = input_shape.rank();
+    output_shape.rank(n);
+    for (int32_t ni = 0; ni < n; ++ni)
+    {
+      output_shape.dim(ni).unset();
+    }
+    return output_shape;
+  }
```

Comment:
I'm considering moving this logic into `pad_shape`  since this logic reused by other pad ops.

2 options
1. add if-statement in current `pad_shape`
2. make another function for non-cosnt case like `pad_non_const_shape()`